modules/ensemble/lib/widget/webview/webview.dart

```diff
@@ -50,13 +50,33 @@ class EnsembleWebView extends StatefulWidget
               ensemble.EnsembleAction.from(funcDefinition, initiator: this),
       'onWebResourceError': (funcDefinition) => _controller.onWebResourceError =
           ensemble.EnsembleAction.from(funcDefinition, initiator: this),
+      'headerOverrideRules': (value) => _controller.headerOverrideRules = 
+          parseHeaderRules(value),
       // legacy
       'uri': (value) => _controller.url = Utils.getUrl(value),
       'allowedLaunchSchemes': (value) => _controller.schemes =
           Utils.getList<String>(value) ??
               EnsembleWebViewController.defaultSchemes,
     };
   }
+
+  List<HeaderOverrideRule> parseHeaderRules(YamlList yamlList) {
+    return yamlList.map((rule) {
+      Map<String, String> headers = {};
+      (rule['headers'] as YamlMap).forEach((key, value) {
```

Comment:
by assuming it's a YamlMap, we make it impossible for someone to set this value in js. Please just use Map